### ISSUE TITLE
how to use Unicode in DosBox-X

### ISSUE THREAD

**Issue by Arthur3032** [joncampbell123/dosbox-x#3464]
### Code of Conduct & Contributing Guidelines

- [X] I agree to follow the code of conduct and the contributing guidelines.

### Have you checked that no other similar question(s) already exists?

- [X] I have searched and didn't find any similar issues.

### Question

dear sir,
How to use Unicode Traditional Chinese characters in DosBox-X.
For example: 煊, 亘

DosBox-X ver 0.83.25
Windows 10, 21H2

**Comment by Wengier** [comment]
I don't think this is really a question about Unicode support, but Chinese character support in individual code pages. There is a code page 951 in DOSBox-X which tries to emulate the (hidden) Traditional Chinese code page 951, and characters including 煊 and 亘 are support there in TTF output. For other outputs, if you set ``uao=true`` in code page 951, they can be typed in ETen Chinese system too. Hope this helps.

**Comment by Arthur3032** [comment]
dear sir,
Outside DosBox-x, copy  word of unicode, For example: 煊, 亘
So, in Dosbox-x, clip$ is not read

**Comment by Jookia** [comment]
Did you enable the clip$ support in the menu? See http://dosbox-x.com/wiki/Guide%3AClipboard-support-in-DOSBox%E2%80%90X#_copy_and_paste_text_using_a_dos_device

**Comment by Jookia** [comment]
Enabling the clip$ support and typing 'chcp 951' and 'type clip$' managed to paste that text, however you need to make sure you're using a TTF font that supports it. /usr/share/fonts/truetype/unifont/unifont.ttf on my system supports it.

**Comment by Arthur3032** [comment]
Can you provide traditional Chinese unifont

**Comment by Jookia** [comment]
I think this will work: https://unifoundry.com/pub/unifont/unifont-14.0.03/font-builds/unifont-14.0.03.ttf Download that and set it as the font.

Note that this doesn't include code points outside the basic multilingual plane: https://en.wikipedia.org/wiki/Plane_(Unicode)#Basic_Multilingual_Plane

As far as I'm aware no font includes every unicode code point due to size. DOSBox-X would need to implement more complex font support for that, or you would need to make your own fonts including certain ranges.

An intermediate could be to build and distribute a font for each code page.

**Comment by surphome** [comment]
Hello dear Jookia 
I used educedit.exe to make a lot of characters, but dosbox-x can't display these characters. Is it possible to achieve this?
How wonderful it would be if it could come true

**Comment by surphome** [comment]
@Jookia  print screen comparison ,
![image](https://user-images.githubusercontent.com/105182792/168411204-98daca71-f5cd-43a2-9f93-5a17a011694e.png)

**Comment by Jookia** [comment]
I don't think DOS supports Unicode private use area code points, only other encodings. @Wengier might have more information.

**Comment by surphome** [comment]
Windows 10 has installed the full font library provided by the government. It is possible to input rare fonts in word and notepad, but dosbox-x cannot input these rare words. It is a pity. I look forward to the support of experts to make this happen and benefit the public. Thank you

**Comment by rderooy** [comment]
I'm no expert on the subject, but regular MS-DOS or IBM PC DOS is limited to 256 characters in total. Therefore the need for codepages depending on region as 256 characters is not sufficient to even handle all the Latin characters in addition to math and things like box drawing characters. Never mind when you get into writing systems other then Latin.

The only way to get different characters in DOS is to load a codepage that contains them. This effects both input and what is presented onscreen, unless the program draws the characters in a graphical mode.

This also effects saving and loading simple text files. If you save a file, and then change codepage and load the file, depending on the characters that were used in the file, it may appear like garbage.

Loading a unicode codepage is simply not possible, and even if it were, no DOS program would know what to do with it.

I know that for Asian countries there is a DBCS (double byte character set) for DOS/V which should allow more characters, but I don't know much about that.

**Comment by surphome** [comment]
Thank you for your reply, I deeply appreciate your enthusiasm

**Comment by Arthur3032** [comment]
In my opinion, in addition to grabbing the glyphs specified by doxbox-x, you have to grab a user-defined font file for Windows.
